src/services/smartWallet.js

```diff
@@ -171,7 +140,15 @@ class SmartWallet {
     const deployEstimate = await this.sdk.estimateAccountDeployment().catch(this.handleError);
 
     const accountBalance = this.getAccountRealBalance();
-    if (get(deployEstimate, 'totalCost') && accountBalance.gte(deployEstimate.totalCost)) {
+    if (get(deployEstimate, 'gasFee')
+      && get(deployEstimate, 'signedGasPrice.gasPrice')
+      && accountBalance.gte(deployEstimate.gasFee.mul(deployEstimate.signedGasPrice.gasPrice))
```

Comment:
Would it make sense to extract this into a function?

Reply:
Maybe it would just unnecessarily add extra complexity

Reply:
Separated, please take a look and resolve if OK.